src/js/angular/repositories/ontop-repo.directive.js

```diff
@@ -0,0 +1,300 @@
+import {getFileName} from "./controllers";
+
+angular
+    .module('graphdb.framework.repositories.ontop-repo.directive', [])
+    .directive('ontopRepo', ontopRepoDirective);
+
+ontopRepoDirective.$inject = ['$modal', 'RepositoriesRestService', 'toastr', 'Upload', 'ModalService'];
+
+function ontopRepoDirective($modal, RepositoriesRestService, toastr, Upload, ModalService) {
+    return {
+        restrict: 'E',
+        scope: false,
+        templateUrl: 'js/angular/repositories/templates/ontop-repo.html',
+
+        link: linkFunc
+    };
+
+    function linkFunc($scope) {
+
+        $scope.selectedDriver = {
+            driverType: "generic",
+            driverName: "Generic JDBC Driver",
+            jdbc: {
+                hostName: "",
+                port: "",
+                databaseName: "",
+                userName: "",
+                password: "",
+                driverClass: "",
+                url: ""
+            },
+            urlTemplate: "",
+            downloadDriverUrl: ""
+        };
+        $scope.ontopRepoFileNames = {};
+        $scope.supportedDriversData = [];
+        $scope.ontopRepoFileLabels =
+            {propertiesFile: 'JDBC properties', obdaFile: 'OBDA or R2RML', owlFile: 'ontology file', constraintFile: 'constraint file'};
+        $scope.ontopRepoFiles = _.keys($scope.ontopRepoFileLabels);
```

Comment:
No need to use lodash for this, just replace with `Object.keys($scope.ontopRepoFileLabels);`